### PR TITLE
fix(atomic, quantic): broken HTML because of formatting in CRGA markdown heading

### PR DESCRIPTION
**Pull request by lbergeron**
[SVCC-4245](https://coveord.atlassian.net/browse/SVCC-4245)

A CRGA model generally generates markdown headings like this:
```
# Title
```

However, in some cases, it can generate headings this way:
```
**Title**
=====
```

When it does, the HTML returned by the `generated-answer` component is broken and the heading reads like this:
```
Title>Title
```

The core of the issue is in the custom Marked renderer. When the heading has formatting, the `text` parameter contains the HTML rendering of the heading content. Inserting non-escaped HTML into the `aria-label` attribute then breaks the `div` element.

This PR fixes the issue by affecting only the heading text content to the `aria-label` attribute. To do so, the HTML elements are removed from the heading content when setting `aria-label`. The heading content HTML is kept intact when setting the heading inner HTML though in order to keep the heading formatting.

For example:
```
**Title**
=====
```
Would generate this HTML
```
<div part="answer-heading-1" aria-label="Title">
  <strong part="answer-strong">Title</strong>
</div>
```

[SVCC-4245]: https://coveord.atlassian.net/browse/SVCC-4245?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/atomic/src/components/common/generated-answer/generated-content/markdown-utils.test.ts [hunk 1]
```diff
@@ -302,6 +302,22 @@ describe('markdownUtils', () => {
...
+
+      it('should transform formatted heading', () => {
+        const text = '# **bold** *emphasized* with `code` title';
+        const html = transformMarkdownToHtml(text);
+        expect(removeLineBreaks(html)).toBe(
+          '<div part="answer-heading-1" aria-label="bold emphasized with code title"><strong part="answer-strong">bold</strong> <em part="answer-emphasis">emphasized</em> with <code part="answer-inline-code">code</code> title</div>'
+        );
+      });
+
+      it('should transform heading with nested formatting', () => {
```
Comment (SimonMilord) [inline packages/atomic/src/components/common/generated-answer/generated-content/markdown-utils.test.ts:314]: should we also have this test for nested formatting in the quantic tests?

**Review by SimonMilord — APPROVED** [review]

File: packages/atomic/src/components/common/generated-answer/generated-content/markdown-utils.test.ts (hunk 1 above)
Comment (lbergeron) [reply packages/atomic/src/components/common/generated-answer/generated-content/markdown-utils.test.ts:314]: Considering the test suite was simplified for Quantic, I went with a single test to cover the change. I don't think adding another test for nested formatting is necessary.

**Review by alexprudhomme — CHANGES_REQUESTED** [review]
Hi, because of the incident that occured last thursday/friday with the v2 release. It wrongfully updated the versions of all package.json in the `ui-kit publish` commit. We removed that commit after that PR pulled it. Can you revert those changes ?

**Comment by lbergeron** [comment]
Creating a clean PR with the TIP of `v2`. See https://github.com/coveo/ui-kit/pull/4564

**Pull request closed without merging**